src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/ByteArrayConverter.cs

```diff
@@ -29,6 +33,176 @@ public override void Write(Utf8JsonWriter writer, byte[]? value, JsonSerializerO
             }
         }
 
+        internal override bool OnTryRead(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options, scoped ref ReadStack state, out byte[]? value)
+        {
+            if (state.Current.ObjectState < StackFrameObjectState.CreatedObject)
```

Comment:
Like with existing resumable converters, there should be a fast path that runs when `SupportContinuation` is set to false. In this case we can assume that the string has been fully populated and therefore we can read it using the non-segmenting APIs.

Reply:
This is handled in `JsonConverter.TryRead` where we check `ConverterStrategy == ConverterStrategy.SegmentableValue && !state.IsContinuation && !reader._hasPartialStringValue` which guarantees that only partially populated buffers will go through this path since fully populated buffers would have `reader._hasPartialStringValue == false`. It's too late to do the check here since the stack already is pushed at this point so it needs to be in `JsonConverter.TryRead` or earlier.

Reply:
`JsonConverter.TryRead` is a hot path method and historically changing it has been causing a great deal of performance regressions. I would recommend moving as much of the logic as possible inside the resumable converters themselves. I would also recommend keeping as much consistency as possible with the existing resumable converters, since their code is extremely nontrivial and difficult to follow.